packages/edit-site/src/components/block-editor/use-site-editor-settings.js

```diff
@@ -87,6 +90,18 @@ function useArchiveLabel( templateSlug ) {
 	);
 }
 
+function useGoBack() {
+	const location = useLocation();
+	const history = useHistory();
+	const goBack = useMemo( () => {
+		const isFocusMode =
+			location.params.focusMode ||
+			FOCUSABLE_ENTITIES.includes( location.params.postType );
```

Comment:
I think it's possible that the url has postType but no postId in which case, we shouldn't have a goBack.

Reply:
Also, I believe we should be checking the "previous" location, not the current one.

Reply:
I'm reusing the existing logic we have for showing the back button:

https://github.com/WordPress/gutenberg/pull/58528/files/d6f6546c033cca1b56469dcecc8efa75018c0c6d#diff-f1dce6f7604716e169988e95b1381324f69e43bb718b523ce79d3dfc2d1ef966L28-L32

What do you mean by checking the previous location?

Reply:
Taking a stab at what you mean but also sharing my idle thoughts 🙂

It would be a nicer flow if we only showed the Back button when you get there by e.g. selecting a pattern block and clicking _Edit_ and **not** when you get there using the W menu. I'll see if it's possible. I don't think we currently keep track of the entity that you were previously editing.

Reply:
> It would be a nicer flow if we only showed the Back button when you get there by e.g. selecting a pattern block and clicking Edit and not when you get there using the W menu. I'll see if it's possible. I don't think we currently keep track of the entity that you were previously editing.

This is basically what I'm saying: the "back button" should only be visible if the previous page is a non focused entity (so postId and postType) defined but also different than the current entity.

Reply:
Let's handle this in a follow-up. I think we'd need to make changes to `@wordpress/router` which deserve some consideration. Right now the logic controlling when the back button appears is the same as it is in `trunk`, just the button is in a different place.

Reply:
This was simpler than I thought: https://github.com/WordPress/gutenberg/pull/58710 TIL about `usePrevious` 😅